src/index.rs

```diff
@@ -512,6 +512,16 @@ impl Index {
     }
   }
 
+  pub(crate) fn list_unspent(&self, outpoint: OutPoint) -> Result<Vec<(u64, u64)>> {
```

Comment:
I think this would be better placed in the `wallet` module, and be combined with the `list_unspent` function there.